WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerActivity.java

```diff
@@ -104,36 +151,64 @@ public void onEventMainThread(CoreEvents.BlogListChanged event) {
         }
     }
 
-    private boolean hasAdapter() {
-        return mAdapter != null;
-    }
-
     private SitePickerAdapter getAdapter() {
         if (mAdapter == null) {
-            mAdapter = new SitePickerAdapter(this);
+            mAdapter = new SitePickerAdapter(this, mCurrentLocalId);
             mAdapter.setOnSiteClickListener(this);
-            mAdapter.setOnMultiSelectedListener(this);
+            mAdapter.setOnSelectedCountChangedListener(this);
         }
         return mAdapter;
     }
 
     @Override
-    public void onMultiSelectEnabled() {
-        if (mActionMode == null) {
-            startSupportActionMode(new ActionModeCallback());
+    public void onSelectedCountChanged(int numSelected) {
+        if (mActionMode != null) {
             updateActionModeTitle();
+            mActionMode.invalidate();
         }
     }
 
-    @Override
-    public void onSelectedCountChanged(int numSelected) {
-        if (mActionMode != null) {
-            if (numSelected == 0) {
-                mActionMode.finish();
-            } else {
-                updateActionModeTitle();
-                mActionMode.invalidate();
+    private void saveHiddenSites() {
+        WordPress.wpDB.getDatabase().beginTransaction();
+        try {
+            // make all sites visible...
+            WordPress.wpDB.setAllDotComBlogsVisibility(true);
+
+            // ...then update ones marked hidden in the adapter, but don't hide the current site
+            boolean skippedCurrentSite = false;
+            String currentSiteName = null;
+            SiteList hiddenSites = getAdapter().getHiddenSites();
+            for (SiteRecord site : hiddenSites) {
+                if (site.localId == mCurrentLocalId) {
+                    skippedCurrentSite = true;
+                    currentSiteName = site.getBlogNameOrHostName();
+                } else {
+                    WordPress.wpDB.setDotComBlogsVisibility(site.localId, false);
+                }
+            }
+
+            // let user know the current site wasn't hidden
+            if (skippedCurrentSite) {
+                ToastUtils.showToast(this,
+                        getString(R.string.site_picker_cant_hide_current_site, currentSiteName),
+                        ToastUtils.Duration.LONG);
             }
+
+            WordPress.wpDB.getDatabase().setTransactionSuccessful();
+        } finally {
+            WordPress.wpDB.getDatabase().endTransaction();
+        }
+    }
+
+    private void addSite() {
+        // if user is signed into wp.com use the dialog to enable choosing whether to
+        // create a new wp.com blog or add a self-hosted one
+        if (AccountHelper.isSignedIn()) {
```

Comment:
`AccountHelper.isSignedIn()` returns true is the user is signed in wpcom *or* in a self hosted (worth a comment in AccountHelper.java).

This test should check if `AccountHelper.getDefaultAccount().isWordPressComUser()` is true.